internal/characters/xianyun/plunge.go

```diff
@@ -0,0 +1,249 @@
+func init() {
+	driftcloudFrames = make([][]int, 3)
+	// skill (press) -> high plunge -> x
+	driftcloudFrames[0] = frames.InitAbilSlice(65) // max
+	driftcloudFrames[0][action.ActionAttack] = 57
+	driftcloudFrames[0][action.ActionCharge] = 56 + 7 // Windup 7 frames
+	driftcloudFrames[0][action.ActionSkill] = 56
+	driftcloudFrames[0][action.ActionBurst] = 54
+	driftcloudFrames[0][action.ActionDash] = 51
+	driftcloudFrames[0][action.ActionJump] = 56
+	driftcloudFrames[0][action.ActionSwap] = 49
+
+	driftcloudFrames[1] = frames.InitAbilSlice(70) // max
+	driftcloudFrames[1][action.ActionAttack] = 60
+	driftcloudFrames[1][action.ActionCharge] = 61 + 7 // Windup 7 frames
+	driftcloudFrames[1][action.ActionSkill] = 55
+	driftcloudFrames[1][action.ActionBurst] = 61
+	driftcloudFrames[1][action.ActionDash] = 55
+	driftcloudFrames[1][action.ActionJump] = 62
+	driftcloudFrames[1][action.ActionSwap] = 53
+
+	driftcloudFrames[2] = frames.InitAbilSlice(76) // max
+	driftcloudFrames[2][action.ActionAttack] = 66
+	driftcloudFrames[2][action.ActionCharge] = 67 + 7 // Windup 7 frames
+	driftcloudFrames[2][action.ActionSkill] = 64
+	driftcloudFrames[2][action.ActionBurst] = 67
+	driftcloudFrames[2][action.ActionDash] = 63
+	driftcloudFrames[2][action.ActionJump] = 68
+	driftcloudFrames[2][action.ActionSwap] = 62
+
+	// high_plunge -> x
+	highPlungeFramesXY = frames.InitAbilSlice(68)
+	highPlungeFramesXY[action.ActionAttack] = 59
+	highPlungeFramesXY[action.ActionCharge] = 59 + 5 // Windup 5 frames
+	highPlungeFramesXY[action.ActionSkill] = 59
+	highPlungeFramesXY[action.ActionBurst] = 59 // Assumed to be the same as skill
+	highPlungeFramesXY[action.ActionDash] = 46
+	highPlungeFramesXY[action.ActionWalk] = 67
+	highPlungeFramesXY[action.ActionSwap] = 51
+
+	// low_plunge -> x
+	lowPlungeFramesXY = frames.InitAbilSlice(65)
+	lowPlungeFramesXY[action.ActionAttack] = 56
+	lowPlungeFramesXY[action.ActionCharge] = 57 + 7 // Windup 7 frames
+	lowPlungeFramesXY[action.ActionSkill] = 59
+	lowPlungeFramesXY[action.ActionBurst] = 59 // Assumed to be the same as skill
+	lowPlungeFramesXY[action.ActionDash] = 44
+	lowPlungeFramesXY[action.ActionSwap] = 48
+}
+
+func (c *char) HighPlungeAttack(p map[string]int) (action.Info, error) {
+	// last action must be skill (for leap)
+	// dont need to check airborne for this because she can plunge if she's on the ground anyways
+	if c.StatusIsActive(skillStateKey) {
+		return c.driftcloudWave(p)
+	}
+
+	switch c.Core.Player.Airborne() {
+	case player.AirborneVenti:
+		return action.Info{}, fmt.Errorf("%s high_plunge while airborne due to venti is unimplemented due to lack of frame data. Please see https://docs.gcsim.app/mechanics/frames for how to contribute", c.Base.Key.String())
+	case player.AirborneXianyun:
+		return c.highPlungeXY(p)
+	default:
+		return action.Info{}, fmt.Errorf("%s high_plunge can only be used while airborne", c.Base.Key.String())
+	}
+}
+
+func (c *char) driftcloudWave(_ map[string]int) (action.Info, error) {
```

Comment:
don't need to pass in the param map here if you're not using it. there's no signature requirement on this func